purchase_stock_price_unit_sync/models/purchase_order.py

```diff
@@ -10,8 +10,11 @@ class PurchaseOrderLine(models.Model):
 
     def write(self, vals):
         res = super().write(vals)
-        if ("price_unit" in vals or "discount" in vals) and (
-            not self.env.context.get("skip_stock_price_unit_sync")
+        if (
+            ("price_unit" in vals or "discount" in vals)
+            and not self.env.context.get("skip_stock_price_unit_sync")
+            # This context is present when the purchase_discount hack is being made
+            and not self.env.context.get("purchase_discount")
```

Comment:
I don't find this context in `purchase_discount` v15 code.

Reply:
Ok, changed to `skip_update_price_unit` (according to `purchase_discount` v15 code).